### PR TITLE
Refactor last minor class components

### PR DESCRIPTION
**Pull request by EjPlatzer**
This PR refactors the last small, self-contained class components into function components.

I also took the time to do some minor refactoring of the affected components, and sometimes related components when it made sense (i.e. properly extracting shared styles).

### CODE REVIEW

File: src/services/session.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ type Session = {
   SessionEndDate: string;
 };
 
-type DaysLeft = [daysRemaining: number, daysCompleted: number];
+type DaysLeft = [daysRemaining: number, daysInSemester: number];
```
Comment (bennettforkner) [inline src/services/session.ts:10]: Has this been changed in the API? Was it named incorrectly before?

**Review by bennettforkner — CHANGES_REQUESTED** [review]

File: src/services/session.ts (hunk 1 above)
Comment (EjPlatzer) [reply src/services/session.ts:10]: So this is a JS "tuple", aka an array. So the names are only for TypeScript/Intellisense reference, the elements of the array aren't actually named. Also, this is the correct name, matching the one in the API. It was incorrect before but inconsequential.

Comment (bennettforkner) [reply src/services/session.ts:10]: Oh yes, thanks for clarifying for me. Great!

**Review by bennettforkner — APPROVED** [review]
LGTM!

**Pull request merged**